libtransmission/rpc-server.cc

```diff
@@ -638,6 +638,17 @@ static void handle_request(struct evhttp_request* req, void* arg)
             return;
         }
 
+        evhttp_add_header(req->output_headers, "Access-Control-Allow-Origin", "*");
+
+        if (req->type == EVHTTP_REQ_OPTIONS)
+        {
+            char const* headers = evhttp_find_header(req->input_headers, "Access-Control-Request-Headers");
+            evhttp_add_header(req->output_headers, "Access-Control-Allow-Methods", "GET, POST, OPTIONS");
+            evhttp_add_header(req->output_headers, "Access-Control-Allow-Headers", headers);
```

Comment:
Looks reasonable, but do we know that `headers` will always be non-nullptr here? Looks like `evhttp_add_header()` does not check for nullptr before using it, so we might want to add a safety guard here.